### PR TITLE
Deploy intro

### PR DESCRIPTION
**Pull request by Hoverbear**
Refactors the deploy intro page

* Simplifies system requirements discussion and removes a lot of verbiage.
* Clearly lays out testing vs production vs dev requirements without repeating ourselves.
* Shuffles folks off to relevant pages.

### CODE REVIEW

**Comment by Hoverbear** [comment]
PTAL @dcalvin

File: content/docs/3.0/tasks/deploy/introduction.md [hunk 1]
```diff
@@ -7,55 +7,66 @@ menu:
...
-|         |         |         |              | Total Server Number |  4      |
+* A control machine (it can be one of your target servers) with [Ansible](https://www.ansible.com/) installed.
+* Several (6+) machines fulfilling the [system requirements](#system-requirements) and at least up to [production specifications](#production-specifications).
+* The ability to configure your infrastructure to allow the ports from [network requirements](#network-requirements).
 
-> **Note**: 
-> 
-> - Do not deploy PD and TiKV on the same server.
-> - For performance-related test, do not use low-performance storage and network hardware configuration, in order to guarantee the correctness of the test result.
+If you have these, follow through the [Ansible deployment](../ansible) guide. You may optionally choose unsupported manual [Docker deployment](../docker) or [binary deployment](../binary) strategies.
```
Comment (dcalvin) [inline content/docs/3.0/tasks/deploy/introduction.md]:
```suggestion
If you have your production environment ready, follow through the [Ansible deployment](../ansible) guide. You may optionally choose unsupported manual [Docker deployment](../docker) or [binary deployment](../binary) strategies.
```

File: content/docs/3.0/tasks/deploy/introduction.md [hunk 2]
```diff
@@ -7,55 +7,66 @@ menu:
...
 
-### Production environment
+Finally, if you want to **develop** TiKV you should consult the [README](https://github.com/tikv/tikv/blob/master/README.md) of the repository.
```
Comment (dcalvin) [inline content/docs/3.0/tasks/deploy/introduction.md]: I am not sure whether we should include this line - there seems to be no direct relation between TiKV development and TiKV deployment.

File: content/docs/3.0/tasks/deploy/introduction.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hoverbear) [reply content/docs/3.0/tasks/deploy/introduction.md]: As far as I am aware we only officially support Ansible and Kubernetes right now. Maybe @c4pt0r can enlighten us?

File: content/docs/3.0/tasks/deploy/introduction.md [hunk 3]
```diff
@@ -7,55 +7,66 @@ menu:
...
-> - A large number of TiKV tests have been run on the CentOS 7.3 system, and in our community there are a lot of best practices in which TiKV is deployed on the Linux operating system. Therefore, it is recommended to deploy TiKV on CentOS 7.3 or later.
-> - The support for the Linux operating systems above includes the deployment and operation in physical servers as well as in major virtualized environments like VMware, KVM and XEN.
+For your **first steps** into TiKV requires only a few things:
 
-## Server requirements
+* A modest machine that fulfills the [system requirements](#system-requirements).
+* A running [Docker](https://docker.com) service.
 
-You can deploy and run TiKV on the 64-bit generic hardware server platform in the Intel x86-64 architecture. The requirements and recommendations about server hardware configuration for development, test and production environments are as follows:
+If you have those, follow through the [`docker-compose`](../docker-compose) guide to get a test setup of TiKV running on your machine.
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply content/docs/3.0/tasks/deploy/introduction.md]: `docker-compose` doesn't require a server though, just a machine.

[the remark this answers is not in the document]
Comment (dcalvin) [reply content/docs/3.0/tasks/deploy/introduction.md]: Good to know, what about "the environment" instead. In user docs I am inclined to use more specific references than pronouns such as these and those.

File: content/docs/3.0/tasks/deploy/introduction.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dcalvin) [reply content/docs/3.0/tasks/deploy/introduction.md]: PTAL @c4pt0r If we don't support Docker deployment or binary then we shouldn't redirect users to it.

[the remark this answers is not in the document]
Comment (Hoverbear) [reply content/docs/3.0/tasks/deploy/introduction.md]: @dcalvin It's OK to have guides that are not (currently) supported. We will eventually need to have guides (or else folks will just wing it and make terrible mistakes), so starting from something is better than starting from scratch.

File: content/docs/3.0/tasks/deploy/introduction.md (hunk 2 above)
Comment (Hoverbear) [reply content/docs/3.0/tasks/deploy/introduction.md]: Let's change it to say "If you want to build your own version of TiKV" as folks may want builds using different feature flags which we don't distribute.

**Pull request merged**